src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let rollingState = { ...state };
+
+  for (const action of actions) {
+    switch (action['type']) {
```

Comment:
do you really need this `[]` brackets ?
you can use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) before switch and then use properties